### PR TITLE
fix(bundler): cover recipe.yaml with bundle checksums

### LINKED ISSUE #1549: Bundle recipe.yaml is not covered by checksums.txt / attestation
## Summary

A generated bundle's `recipe.yaml` is **not** covered by `checksums.txt`, and therefore not transitively pinned by the bundle attestation (which signs the digest of `checksums.txt`). It can be modified after generation while `aicr verify` still reports success.

## Evidence

- The Helm deployer writes `checksums.txt` in `pkg/bundler/deployer/helm/helm.go:229-233`.
- `runDeployer` writes `recipe.yaml` **after** that, in `pkg/bundler/bundler.go:514-525`.
- So `recipe.yaml` is absent from `checksums.txt`; tampering with it leaves `aicr verify` passing.

Surfaced while reviewing PR #1547 (the bundle-attestation demo claims `checksums.txt` "pins every content file", which is currently inaccurate for `recipe.yaml`).

## Fix

Include `recipe.yaml` in the checksum set (write it before checksum computation, or add it to the manifest), and add a regression test asserting `recipe.yaml` is listed in `checksums.txt` and that tampering it fails verification. Until then, #1547 carries an explicit caveat.

### PR DESCRIPTION
**Pull request by yuanchen8911**
## Summary

Include the resolved Helm bundle `recipe.yaml` in `checksums.txt` by writing and tracking it before checksum finalization. Add generated-bundle tamper regression coverage and update the documented attestation boundary.

## Motivation / Context

Helm bundles wrote `recipe.yaml` after `checksums.txt`, so modifying the resolved recipe did not fail `aicr verify` and was not transitively pinned by the bundle attestation.

Fixes #1549
Related: #1547

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Build/CI/tooling

## Component(s) Affected

- [ ] CLI (`cmd/aicr`, `pkg/cli`)
- [ ] API server (`cmd/aicrd`, `pkg/server`)
- [ ] Recipe engine / data (`pkg/recipe`)
- [x] Bundlers (`pkg/bundler`, `pkg/component/*`)
- [ ] Collectors / snapshotter (`pkg/collector`, `pkg/snapshotter`)
- [ ] Validator (`pkg/validator`)
- [ ] Core libraries (`pkg/errors`, `pkg/k8s`)
- [x] Docs/examples (`docs/`, `examples/`)
- [ ] Other: ____________

## Implementation Notes

- Serialize the resolved Helm recipe before deployer generation.
- Pass the recipe path to the Helm generator so it participates in the generated file list, size accounting, checksum manifest, and downstream attestation chain.
- Verify an actual generated bundle lists `recipe.yaml` and rejects post-generation recipe tampering.
- Remove the temporary `#1549` caveats from user, integrator, and demo documentation.

Non-Helm deployer layouts remain unchanged because they do not emit a resolved `recipe.yaml` today.

## Testing

```bash
GOFLAGS="-mod=vendor" go test -run '^TestMake_RecipeCoveredByChecksums$' ./pkg/bundler
GOFLAGS="-mod=vendor" go test ./pkg/bundler/...
GOFLAGS="-mod=vendor" go test -coverprofile=/tmp/aicr/issue-1549-cover.out ./pkg/bundler/...
golangci-lint run --timeout=10m -c .golangci.yaml ./pkg/bundler/...
unset GITLAB_TOKEN && make test-coverage
unset GITLAB_TOKEN && make qualify
```

All commands passed. Combined `pkg/bundler/...` coverage remained 80.3% (80.3% baseline to 80.3% current), while `pkg/bundler/deployer/helm` increased from 83.7% to 84.2%. Project-wide coverage passed at 78.8% against the 75% floor. Qualification passed race tests, vet, lint, repository consistency checks, all 23 Chainsaw e2e suites, vulnerability scanning, and license checks.

## Risk Assessment

- [x] **Low** — Isolated change, well-tested, easy to revert
- [ ] **Medium** — Touches multiple components or has broader impact
- [ ] **High** — Breaking change, affects critical paths, or complex rollout

**Rollout notes:** Backward compatible. Newly generated Helm bundles bind `recipe.yaml` into `checksums.txt`; existing bundles retain their original checksum scope.

## Checklist

- [x] Tests pass locally (`make test` with `-race`)
- [x] Linter passes (`make lint`)
- [x] I did not skip/disable tests to make CI green
- [x] I added/updated tests for new functionality
- [x] I updated docs if user-facing behavior changed
- [x] Changes follow existing patterns in the codebase
- [x] Commits are cryptographically signed (`git commit -S`) — [GPG signing info](https://docs.github.com/en/authentication/managing-commit-signature-verification)

### CODE REVIEW

File: pkg/bundler/bundler.go [hunk 1]
```diff
@@ -567,6 +573,15 @@ func (b *DefaultBundler) argoDeployerOptions() (*config.ArgoDeployerOptions, err
...
 func (b *DefaultBundler) runDeployer(ctx context.Context, d deployer.Deployer, recipeResult *recipe.RecipeResult, dir string, dataFiles []string, start time.Time) (*result.Output, error) {
+	// Helm bundles include the resolved recipe. Write it before generation so
+	// the Helm deployer can include it in checksums.txt and the resulting
+	// attestation chain.
+	if b.Config.Deployer() == config.DeployerHelm {
+		if _, writeErr := b.writeRecipeFile(recipeResult, dir); writeErr != nil {
+			return nil, errors.Wrap(errors.ErrCodeInternal, "failed to write recipe file", writeErr)
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (yuanchen8911) [reply pkg/bundler/bundler.go:584]: Fixed in 54a50cc8. I verified that writeRecipeFile already returns structured errors, so runDeployer now uses errors.PropagateOrWrap to preserve those codes and avoid duplicate context. Validation: bundler race tests passed, affected-package lint reported 0 issues, and the final clean-cache make qualify run passed (78.8% coverage, 23/23 e2e).

**Comment by yuanchen8911** [comment]
CI note for the latest head (`54a50cc8`): the red `tests / Test` and `tests / CLI E2E` jobs fail during action preparation, before AICR tests run. NVIDIA’s enterprise Actions policy rejects the pinned `anchore/sbom-action`, `goreleaser/goreleaser-action`, and `sigstore/cosign-installer` actions. Local `make qualify` passes in full (78.8% coverage, lint 0 issues, 23/23 e2e).

**Review by mchmarny — APPROVED** [review]
Solid supply-chain fix. The write-before-generate reordering is correct: `RecipeFile` is set only on the `DeployerHelm` branch of `buildDeployer`, matching exactly where `runDeployer` writes the file, so no other deployer can hit the missing-file path. `AddDataFiles` preserves the old file-count/size accounting, goes through `SafeJoin`, and fails closed when the recipe is absent — pinned by `TestGenerate_MissingRecipeFile`. Most importantly, `TestMake_RecipeCoveredByChecksums` exercises the real `Make` → tamper → `Verify` path and asserts the failure identifies `recipe.yaml`, which directly guards the false-PASS direction this bug enabled. The thorough caveat cleanup across docs and demos matches the new attestation boundary. CI green, no blockers.

**Review by njhensley — APPROVED** [review]
**Multi-persona review + adversarial verification — Approve** ✅

Reviewed by an independent 3-lens panel (supply-chain correctness · bundler-flow correctness · test-coverage), then the ordering, exactly-once-accounting, and scope-completeness claims were re-derived from the code by an adversarial meta-reviewer.

The ordering crux is safe: `recipe.yaml` is written to the already-`MkdirAll`'d output dir before `helm.Generate`, and generation never removes it — `pruneStaleFolders` skips non-directories and only `RemoveAll`s strict `NNN-` folders; the only other deletion is a targeted `os.Remove(undeploy.sh)`. It's counted **exactly once** via `output.AddDataFiles` (the old `totalFiles++/totalSize+=recipeSize` block is fully removed, the write's return discarded). Scope is complete — `writeRecipeFile` is the only writer of `recipe.yaml` repo-wide and runs only under `DeployerHelm`; non-Helm bundles emit no recipe, so nothing unchecksummed leaks. `RecipeFile` is set only inside the Helm `Generator` literal (not a shared options struct). Fail-closed by design: a future reorder that writes after generation errors loudly (`failed to stat data file`), locked in by `TestGenerate_MissingRecipeFile`. The tamper regression (`TestMake_RecipeCoveredByChecksums`) exercises the real Make→Verify path. All #1549 doc caveats removed (zero `git grep 1549` hits).

Non-blocking nitpicks (🔵):

- 🔵 `writeRecipeFile` still returns `(int64, error)` but the size is now discarded at the sole call site — dead return value, could simplify to `error`-only.
- 🔵 Theoretical double-entry if a user passes `--data` a file literally named `recipe.yaml` (listed twice in `checksums.txt`) — harmless to verification, pre-existing-shaped corner case.
- 🔵 No direct `--attest`-path tamper test — covered transitively (attestation subject is the `checksums.txt` digest, which is tested); an explicit attest-tamper test would be defense-in-depth.

No blockers, majors, or minors.
